Content.Server/GameObjects/Components/Explosion/ClusterFlashComponent.cs

```diff
@@ -0,0 +1,186 @@
+        {
+            if (_countDown || ((_grenadesContainer.ContainedEntities.Count) <= 0 && (_fillPrototype == null)))
+                return false;
+            Owner.SpawnTimer((int) (_delay * 1000), () =>
+            {
+                if (Owner.Deleted)
+                    return;
+                _countDown = true;
+                var random = IoCManager.Resolve<IRobustRandom>();
+                var delay = 20;
+                var grenadesWasInserted = _grenadesContainer.ContainedEntities.Count + _unspawnedCount;
+                var throwedCount = 0;
+                var segmentAngle = (int) (360 / grenadesWasInserted);
+                while (TryGetGrenade(out var grenade))
+                {
+                    var angleMin = segmentAngle * throwedCount;
+                    var angleMax = segmentAngle * (throwedCount + 1);
+                    var angle = Angle.FromDegrees(random.Next(angleMin, angleMax));
+                    var distance = (float)random.NextDouble() * _throwDistance;
+                    var target = new EntityCoordinates(Owner.Uid, angle.ToVec().Normalized * distance);
+
+                    grenade.Throw(0.5f, target, grenade.Transform.Coordinates);
+
+                    grenade.SpawnTimer(delay, () =>
+                    {
+                        if (grenade.Deleted)
+                            return;
+
+                        if (grenade.TryGetComponent(out OnUseTimerTriggerComponent? useTimer))
+                        {
+                            useTimer.Trigger(eventArgs.User);
+                        }
+                    });
+
+                    delay += random.Next(550, 900);
+                    throwedCount++;
+                }
+
+                Owner.Delete();
+            });
+            return true;
+        }
+
+        private bool TryGetGrenade([NotNullWhen(true)] out IEntity? grenade)
+        {
+            grenade = null;
+
+            if (_unspawnedCount > 0)
+            {
+                _unspawnedCount--;
+                grenade = Owner.EntityManager.SpawnEntity(_fillPrototype, Owner.Transform.Coordinates);
+                return true;
+            }
+
+            if (_grenadesContainer.ContainedEntities.Count > 0)
+            {
+                grenade = _grenadesContainer.ContainedEntities[0];
+
+                // This shouldn't happen but you never know.
+                if (!_grenadesContainer.Remove(grenade))
+                    return false;
+
+                return true;
+            }
+
+            return false;
+        }
+
+        private void UpdateAppearance()
+        {
+            if (!Owner.TryGetComponent(out AppearanceComponent? appearance)) return;
+
+            try
+            {
+                appearance.SetData(ClusterFlashVisuals.GrenadesCounter, Convert.ToByte(_grenadesContainer.ContainedEntities.Count + _unspawnedCount));
+                appearance.SetData(ClusterFlashVisuals.GrenadesMax, Convert.ToByte(_maxGrenades));
```

Comment:
you don't need to send maxgrenades everytime. imo, just don't send maxgrenades & dont check the max in the visualizer